server/src/main/java/org/opensearch/cluster/decommission/DecommissionController.java

```diff
@@ -267,4 +278,99 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
             }
         });
     }
+
+    void setRoutingWeights(
+        String awarenessAttributeName,
+        Map<String, Double> weights,
```

Comment:
Lets call this awarenessAttributeToWeightMap?

Reply:
Done.